Welcome message when new member joins Discord

OskoPoland opened the pull request:
Add slash command to test the guild member add command as well as a listener for the guild member add to auto create tagged welcome messages.

leavism commented on package.json:
Any idea why you had to install this version node.js?

OskoPoland replied:
I actually have no idea. I will try and remove it and see if that breaks anything.

leavism commented on src/commands/Bot/emitTest.ts:
Nice work getting the emitter to work! We won't need this in the actual merge though. Remove it from the PR (pull request)

OskoPoland replied:
Will do

leavism commented on src/listeners/guildMemberAddGreeting.ts:
Rather than pointing to the #greet channel via a hard-coded channel ID, take a look at this:
https://old.discordjs.dev/#/docs/discord.js/main/class/Guild?scrollTo=systemChannel

A `systemChannel` is basically the channel in which the server sends the default vanilla welcome messages.

leavism replied:
You almost got it with this recent change. Look at the return type of `guild.systemChannel` provided in the link above. You don't need the `systemChannelID` if you already have a property that gives you the actual `systemChannel` object.

leavism replied:
Hey, you're still hard coding a guild ID. Try getting the guild from the GuildMember object that's passed into the run method.

leavism replied:
This is the one. Nice work!

leavism reviewed (changes requested):
Great work! Look at my reviews and make changes where it's needed.

OskoPoland commented:
I made the changes. The node security package I deleted and it didn't impeded anything so I have no clue. The greet channel message location is no longer hardcoded either instead using the systemChannelID to send the message to the correct channel.

I think the changes show up here I am not sure if I need to make a whole new pull request or if this works as well?

leavism commented:
> I made the changes. The node security package I deleted and it didn't impeded anything so I have no clue. The greet channel message location is no longer hardcoded either instead using the systemChannelID to send the message to the correct channel.
> 
> 
> 
> I think the changes show up here I am not sure if I need to make a whole new pull request or if this works as well?

Commit the new changes to your fork. The PR should automatically show the new commit. Hopefully hahaha

leavism commented on .gitignore:
Ideally you'd just remove the file from your repo entirely because it's still uploaded to your remote repository. No worries if all of this is new and confusing, we are supposed to go over it in our next meeting.

leavism reviewed (changes requested):
I made direct commits to your fork, remember to `git pull` before you begin working on your local fork again. Take a look at what I said about your changes to `GuildMemberAddGreeting` and see if you can refactor your code to be even simpler.

OskoPoland commented:
Ah I see I didn't look hard enough past the ID but I see that it has a TextChannel obj so I simplified the code some more so it's all a one-liner. I also added a listener and message for the removal as well since its practically the same thing.

leavism reviewed (commented):
Nice work

leavism commented:
Remove the `TextChannel` import from `discord.js` and we're good to merge.

leavism reviewed (approved):
LGTM
